jib-core/src/main/java/com/google/cloud/tools/jib/frontend/JavaDockerContextGenerator.java

```diff
@@ -74,31 +74,32 @@
     private final String extractionPath;
 
     private CopyDirective(
-        ImmutableList<Path> sourceFiles, String directoryInContext, String extractionPath) {
-      this.sourceFiles = sourceFiles;
+        ImmutableList<LayerEntry> layerEntries, String directoryInContext, String extractionPath) {
+      this.layerEntries = layerEntries;
       this.directoryInContext = directoryInContext;
       this.extractionPath = extractionPath;
     }
   }
 
   /**
-   * Adds a copy directive for the {@code layerEntry} if it's not empty.
+   * Adds a copy directive for the {@code layerEntries} if it's not empty.
    *
    * @param listBuilder the {@link ImmutableList.Builder} to add to
-   * @param layerEntry the layer entry
+   * @param layerEntries the layer entries
    * @param directoryInContext the directory in the context to put the source files for the layer
+   * @param extractionPath the extraction path to extract the directory to
    */
   private static void addIfNotEmpty(
       ImmutableList.Builder<CopyDirective> listBuilder,
-      LayerEntry layerEntry,
-      String directoryInContext) {
-    if (layerEntry.getSourceFiles().isEmpty()) {
+      List<LayerEntry> layerEntries,
+      String directoryInContext,
+      String extractionPath) {
+    if (layerEntries.isEmpty()) {
       return;
     }
 
     listBuilder.add(
-        new CopyDirective(
-            layerEntry.getSourceFiles(), directoryInContext, layerEntry.getExtractionPath()));
+        new CopyDirective(ImmutableList.copyOf(layerEntries), directoryInContext, extractionPath));
```

Comment:
`getDependencylayerEntries` is already returning `ImmutableList`, so we could have `addIfNotEmtpy` accept `ImmutableList<LayerEntry>`, and we won't need to copy?